api/dps/server_integration_test.go

```diff
@@ -0,0 +1,1024 @@
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+
+		// No data is written in the database, so the index should fail to retrieve anything.
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetTransactionRequest{
+			TransactionID: txID[:],
+		}
+		_, err = server.GetTransaction(context.Background(), req)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIntegrationServer_GetHeightForTransaction(t *testing.T) {
+	transactions := mocks.GenericTransactions(4)
+	txID := transactions[0].ID()
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+		writer := index.NewWriter(db, storage)
+
+		// Insert mock data in database.
+		require.NoError(t, writer.Transactions(mocks.GenericHeight, transactions))
+		require.NoError(t, writer.Close())
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetHeightForTransactionRequest{
+			TransactionID: txID[:],
+		}
+		resp, err := server.GetHeightForTransaction(context.Background(), req)
+
+		require.NoError(t, err)
+		assert.Equal(t, txID[:], resp.TransactionID)
+		assert.Equal(t, mocks.GenericHeight, resp.Height)
+	})
+
+	t.Run("handles indexer failure on HeightForTransaction", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+
+		// No data is written in the database, so the index should fail to retrieve anything.
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.GetHeightForTransactionRequest{
+			TransactionID: txID[:],
+		}
+		_, err = server.GetHeightForTransaction(context.Background(), req)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestIntegrationServer_ListTransactionsForHeight(t *testing.T) {
+	transactions := mocks.GenericTransactions(4)
+	height := mocks.GenericHeight
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		codec, err := zbor.NewCodec()
+		require.NoError(t, err)
+
+		db := helpers.InMemoryDB(t)
+
+		storage := storage.New(codec)
+		reader := index.NewReader(db, storage)
+		writer := index.NewWriter(db, storage)
+
+		// Insert mock data in database.
+		require.NoError(t, writer.Transactions(height, transactions))
+		require.NoError(t, writer.Close())
+
+		server := dps.NewServer(reader, codec)
+
+		req := &dps.ListTransactionsForHeightRequest{
+			Height: mocks.GenericHeight,
+		}
+		resp, err := server.ListTransactionsForHeight(context.Background(), req)
+
+		require.NoError(t, err)
+		assert.Equal(t, mocks.GenericHeight, resp.Height)
+
+		for _, collection := range transactions {
+			wantID := collection.ID()
+			assert.Contains(t, resp.TransactionIDs, wantID[:])
+		}
```

Comment:
As before, we want to check the total length of `resp.TransactionIDs`, to make sure we don't have any extra IDs or duplicates in the ID list.